primitives/constraints/ssz.go

```diff
@@ -32,9 +32,9 @@ type SSZMarshaler interface {
 }
 
 // SSZUnmarshaler is an interface for objects that can be unmarshaled from SSZ format.
-type SSZUnmarshaler[SelfT any] interface {
-	// NewFromSSZ creates a new object from SSZ format.
-	NewFromSSZ(bz []byte) (SelfT, error)
+type SSZUnmarshaler interface {
+	IsUnusedFromSZZ() bool      // some types are not used and we can/must shortcut their marshalling
+	EnsureSyntaxFromSSZ() error // once unmarshalled we will check whether type syntax is correct
```

Comment:
this PR replaces `NewFromSSZ` with a couple of attributes used in the single `sszconstructor.NewFromSSZ`.
That constructor itself has a template like pattens so it will use these methods to handle the instantiation corner cases (right now, just unused type and post serialization checks for a single type)